src/commands/commerce/extension/register.ts

```diff
@@ -0,0 +1,140 @@
+        }),
+        // optional flag
+        'apex-namespace': flags.string({
+            char: 'm',
+            description: msgs.getMessage('extension.register.apexNameSpaceFlagDescription'),
+        }),
+    };
+
+    public org: Org;
+    public statusFileManager: StatusFileManager;
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async run(): Promise<unknown> {
+        this.ux.log(
+            `Retrieving and registering Apex Class ${
+                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                this.flags['apex-classname']
+            }' using username: ${this.org.getUsername()} \n..........`
+        );
+        return {
+            res: this.registerApex(
+                this.flags['registered-extensionname'],
+                this.flags['extensionpointname'],
+                this.flags['apex-classname'],
+                this.org.getUsername()
+            ),
+        };
+    }
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public registerApex(
+        storeRegisteredName: string,
+        storeEPN: string,
+        storeApexClass: string,
+        storeUserName: string
+    ): string {
+        // get apexClass id from table
+        const apexClassId = this.getApexClass(storeApexClass, storeUserName);
+
+        // checks if user passed valid EPN
+        const epnVal = forceDataSoql(
+            `SELECT Value FROM PicklistValueInfo WHERE Value='${storeEPN}' AND EntityParticle.DurableId = 'RegisteredExternalService.ExtensionPointName' LIMIT 1`,
+            storeUserName
+        );
+        if (epnVal.result.totalSize === 0) {
+            throw new SfdxError(msgs.getMessage('extension.register.errEPN'));
+        }
+
+        // Inserts 5 fields into registeredExternalService table
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const results = forceDataRecordCreate(
+            'RegisteredExternalService',
+            `DeveloperName=${storeRegisteredName} MasterLabel=${storeRegisteredName} ExtensionPointName=${storeEPN} ExternalServiceProviderId=${apexClassId} ExternalServiceProviderType='Extension'`,
+            storeUserName
+        );
+        // Validates/checks for already existing unique name?
+        if (results instanceof SfdxError && results.message.indexOf('DUPLICATE_VALUE') > 0) {
+            throw new SfdxError(msgs.getMessage('extension.register.nameAlreadyExists', [storeRegisteredName]));
+        }
+        // JSON response of inserted record
+        return this.getInsertedRecord(storeRegisteredName, storeApexClass, storeUserName);
+    }
+
+    // Queries for apexclassId from existing apexClass table
+    private getApexClass(storeApexClass: string, storeUserName: string): string {
+        const QUERY_GET_APEX_CLASS = `SELECT Id FROM ApexClass WHERE Name='${storeApexClass}' LIMIT 1`;
```

Comment:
nit: If this is the only place this query is used then get rid of the constant and use the query directly